src/index.rs

```diff
@@ -904,4 +904,23 @@ mod tests {
         assert!(new2 > old2);
         assert!(new2 > new1);
     }
+
+    #[test]
+    fn should_correctly_update_one_resource() {
+        run_test_and_clean_up(|path| {
+            create_file_at(path.clone(), Some(FILE_SIZE_1), Some(FILE_NAME_1));
+            create_file_at(path.clone(), Some(FILE_SIZE_2), Some(FILE_NAME_2));
+            let mut actual = ResourceIndex::build(path.clone());
+            let mut file_path = path.clone();
+            file_path.push(FILE_NAME_1);
+            let old_id = ResourceId {
+                data_size: FILE_SIZE_1,
+                crc32: CRC32_1,
+            };
+            let path_buf = CanonicalPathBuf::canonicalize(&file_path).unwrap();
+            let update = actual
+                .update_one(path_buf.clone(), old_id)
+                .expect("Should update one resource correctly");
+        });
```

Comment:
In this current scenario, the update should be empty (nothing changed).

Probably reasoning with paths instead of files would be better here:
* If we remove a file with id `id1` and path `path1` then the update is `deleted(id1)`
* If we modify a file by path `path1` so it's new id is `id2` then the update is `deleted(id1), added(id2)`
* If we rename/move a file with `id1` from `path1` to `path2` then the update is `deleted(id1), added(id1)`
Please, double-check.

Reply:
_Notion of a "file" is confusing here because it is combination of path and content, but index works with paths and contents separately. A "resource" is an identified content, a "path" it's current location._